scripts/import_cldr.py

```diff
@@ -286,15 +287,20 @@ def parse_global(srcdir, sup):
         region_code = region.attrib['iso3166']
         region_currencies = []
         for currency in region.findall('./currency'):
+            cur_code = currency.attrib['iso4217']
             cur_start = _parse_currency_date(currency.attrib.get('from'))
             cur_end = _parse_currency_date(currency.attrib.get('to'))
-            region_currencies.append((currency.attrib['iso4217'],
-                                      cur_start, cur_end,
-                                      currency.attrib.get(
-                                          'tender', 'true') == 'true'))
+            cur_tender = currency.attrib.get('tender', 'true') == 'true'
+            # Tie region to currency.
+            region_currencies.append((cur_code, cur_start, cur_end, cur_tender))
+            # Keep a reverse index of currencies to territorie.
+            all_currencies.setdefault(cur_code, []).append(region_code)
         region_currencies.sort(key=_currency_sort_key)
         territory_currencies[region_code] = region_currencies
 
+    # Deduplicate regions and makes them pickable.
+    all_currencies = dict([(code, tuple(set(regions))) for code, regions in all_currencies.items()])
```

Comment:
I had to resort to that unsexy code to make the metadata serialization happy. Changing the code as you proposed ends up with that failure:

```python
____ test_compatible_classes_in_global_and_localedata[babel/global.dat] ____

filename = 'babel/global.dat'

    @pytest.mark.parametrize('filename', [
        'babel/global.dat',
        'babel/locale-data/root.dat',
        'babel/locale-data/en.dat',
        'babel/locale-data/en_US.dat',
        'babel/locale-data/en_US_POSIX.dat',
        'babel/locale-data/zh_Hans_CN.dat',
        'babel/locale-data/zh_Hant_TW.dat',
        'babel/locale-data/es_419.dat',
    ])
    def test_compatible_classes_in_global_and_localedata(filename):
        # Use pickle module rather than cPickle since cPickle.Unpickler is a method
        # on Python 2
        import pickle
    
        class Unpickler(pickle.Unpickler):
    
            def find_class(self, module, name):
                # *.dat files must have compatible classes between Python 2 and 3
                if module.split('.')[0] == 'babel':
                    return pickle.Unpickler.find_class(self, module, name)
                raise pickle.UnpicklingError("global '%s.%s' is forbidden" %
                                             (module, name))
    
        with open(filename, 'rb') as f:
>           return Unpickler(f).load()

tests/test_core.py:318: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/pickle.py:858: in load
    dispatch[key](self)
/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/pickle.py:1090: in load_global
    klass = self.find_class(module, name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_core.Unpickler instance at 0x10e2e9128>, module = '__builtin__', name = 'set'

    def find_class(self, module, name):
        # *.dat files must have compatible classes between Python 2 and 3
        if module.split('.')[0] == 'babel':
            return pickle.Unpickler.find_class(self, module, name)
        raise pickle.UnpicklingError("global '%s.%s' is forbidden" %
>                                    (module, name))
E       UnpicklingError: global '__builtin__.set' is forbidden

tests/test_core.py:315: UnpicklingError
```